### ISSUE TITLE
bg2bw - apparent infinite loop

### ISSUE THREAD

**Issue by keiranmraine** [cancerit/cgpBigWig#9]
bg2bw appears to go into an infinite loop if Y chr is completely absent, but this isn't replicated if you drop another chromosome from the middle of the file

```
bg2bw -i ascat.bg -o ascat.bw -c genome.fa.fai
```

With this `ascat.bw` the job runs apparently infinitely
```
1	13115	11934653	4
1	11935714	11950891	2
1	11952194	16849660	4
1	16851439	17233668	3
1	17235057	120902392	4
1	120904829	121147880	6
1	121153365	249239902	4
2	10586	87473799	4
2	87475462	87608667	3
2	87615478	243187487	4
3	60196	10186447	2
3	10187633	10298890	0
3	10300030	75426855	2
3	75428270	75645376	4
3	75646494	151508116	2
3	151509698	151550421	4
3	151551690	197908615	2
4	11960	191043877	4
5	11881	180790901	4
6	100115	171049557	4
7	24219	74060455	4
7	74061511	76136389	3
7	76137923	159127076	4
8	11773	146303788	4
9	10468	261190	5
9	262206	141146683	4
10	60968	135524334	4
11	87267	134946318	4
12	68656	133841262	4
13	19020094	115108598	4
14	19000059	107288870	4
15	20000537	20894465	5
15	20939450	102520794	4
16	60290	90159555	4
16	90160792	90292766	3
17	827	81185372	4
18	16443	78017073	4
19	226775	59118779	4
20	61097	62962869	4
21	9411409	48119634	4
22	16051248	51240820	4
X	2699554	88453891	4
X	88455130	92139938	3
X	92143669	154929412	4
```

Just adding a final line resolves this, but see later:

```
Y	0	1	0
```

Taking the above original file and removing chr 10 as well in entirety doesn't induce the infinite loop:

```
1	13115	11934653	4
1	11935714	11950891	2
1	11952194	16849660	4
1	16851439	17233668	3
1	17235057	120902392	4
1	120904829	121147880	6
1	121153365	249239902	4
2	10586	87473799	4
2	87475462	87608667	3
2	87615478	243187487	4
3	60196	10186447	2
3	10187633	10298890	0
3	10300030	75426855	2
3	75428270	75645376	4
3	75646494	151508116	2
3	151509698	151550421	4
3	151551690	197908615	2
4	11960	191043877	4
5	11881	180790901	4
6	100115	171049557	4
7	24219	74060455	4
7	74061511	76136389	3
7	76137923	159127076	4
8	11773	146303788	4
9	10468	261190	5
9	262206	141146683	4
11	87267	134946318	4
12	68656	133841262	4
13	19020094	115108598	4
14	19000059	107288870	4
15	20000537	20894465	5
15	20939450	102520794	4
16	60290	90159555	4
16	90160792	90292766	3
17	827	81185372	4
18	16443	78017073	4
19	226775	59118779	4
20	61097	62962869	4
21	9411409	48119634	4
22	16051248	51240820	4
X	2699554	88453891	4
X	88455130	92139938	3
X	92143669	154929412	4
```

**Comment by keiranmraine** [comment]
For my testing after the fact, dbGaP dataset affected:

```
WGS_LP6007492_F28green_TumorF28-3_vs_LP6007457_F28green_Blood
```

**Comment by ghost** [comment]
Fixed in libBigWig 0.3.1 thanks to @dpryan79 for the rapid response.
